### PR TITLE
Updating to OEP-18 standards and latest version of the analytics-data-api-client

### PR DESCRIPTION
**Pull request by azafty468**
Updating to the OEP-18 standard.  As part of this I also took the opportunity to update to the latest version of the analytics-data-api-client as it was about a year out of date.

### CODE REVIEW

**Comment by azafty468** [comment]
Lost track of this guy with the Hackathon, but the code is ready for review.  Tyler, there are some code changes to the core libraries.  All of these changes are incorporating the new naming scheme for the updating Data API client.  There should be no material changes.

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
 COVERAGE = $(ROOT)/build/coverage
 NODE_BIN=./node_modules/.bin
 
 DJANGO_SETTINGS_MODULE ?= "analytics_dashboard.settings.local"
 
-.PHONY: requirements clean
+.PHONY: requirements clean upgrade pin_pip
 
 # pin to 9.0.3 until tox-battery upgrades
+# 9.0.3 does not work on Sierra Mac OSX.  10.0.1 should be used instead.
```
Comment (thallada) [inline Makefile:10]: It looks like [tox-battery supports pip 10 now](https://github.com/signalpillar/tox-battery/pull/15), so we should be able to remove this pin?

**Review by thallada — APPROVED** [review]
This LGTM. Thanks for making it conform to OEP-18 and for upgrading the API client too!

File: Makefile (hunk 1 above)
Comment (jmbowman) [reply Makefile:10]: We're still using pip 9.0.3 in configuration for most services, I'm inclined to stick with that for now if it's working ok.  The initial pip 10 release broke a number of corner cases like installing from GitHub; I think it's mostly fixed by now, but not sure if all of it is.

File: Makefile [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+
+	# Post process all of the files generated above to replace the instructions for recreating them
+	scripts/post-pip-compile.sh \
+		requirements/pip-tools.txt \
+		requirements/base.txt \
+		requirements/doc.txt \
+		requirements/local.txt \
+		requirements/optional.txt \
+		requirements/production.txt \
+		requirements/test.txt
```
Comment (jmbowman) [inline Makefile]: We found a better way to do this recently, so you shouldn't need the `post-pip-compile.sh` script.  Just set the `CUSTOM_COMPILE_COMMAND` environment variable, as used in https://github.com/edx/xqueue/pull/236

File: requirements/production.txt [hunk 3]
```diff
@@ -1,9 +1,66 @@
-# Pro-tip: Try not to put anything here. There should be no dependency in
-#	production that isn't in development.
```
Comment (jmbowman) [inline requirements/production.txt:2]: Do you think it's worth preserving this note in `production.in`?  There should at least be some kind of comment in that file explaining what those requirements are for.

File: requirements/optional.in [hunk 4]
```diff
@@ -0,0 +1 @@
+newrelic==2.98.0.81             # New Relic
```
Comment (jmbowman) [inline requirements/optional.in:3]: Maybe add a comment to the top of the file explaining why this dependency is optional?

File: requirements/local.in [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# Local development dependencies go here
+-r base.txt
```
Comment (jmbowman) [inline requirements/local.in:2]: Should this be derived from `test.txt` instead?  It seems like you'd normally want to be able to run tests in the local dev environment.

**Review by jmbowman — COMMENTED** [review]
Looks pretty good, just a few suggestions.

**Comment by fysheets** [comment]
I believe this can be closed as https://github.com/edx/edx-analytics-dashboard/pull/808 updated the repo to OEP-18.

**Pull request closed without merging**